omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,187 @@
+        #raise KeyError(msg)
+        return False
+
+    # Check if we got several times the same labels
+    for label in LABELS:
+        if len(df.loc[df[LABEL] == label]) > 1:
+            msg = f'Found label {label} several times. Expected only once.'
+            LOG.error(msg)
+            raise KeyError(msg)
+
+    # Validate the columns we need: BETSTAR{X,Y} and ERRBETSTAR{X,Y}
+    expected_columns = [f'{BETASTAR}{p}' for p in PLANES] + \
+                       [f'{ERR}{BETASTAR}{p}' for p in PLANES]
+    if not all([column in df.columns for column in expected_columns]):
+        #msg = 'Expected columns in the TFS file not found. Expected ' \
+        #      f'columns: {expected_columns}'
+        #LOG.error(msg)
+        #raise KeyError(msg)
+        return False
+    
+    return True
+
+
+def get_imbalance(df):
+    rel_error = {}
+    eff_beta = {}
+    for ip in IPS:
+        # Get the individual rows for the IP1 and IP5
+        row = df.loc[df[LABEL].str.startswith(ip)]
+
+        # Compute the average β* for each plane with both beams
+        # and the absolute errors for each plane
+        average_beta = {}
+        error = {}
+        for p in PLANES:
+            average_beta[p] = 0.5 * row[f'{BETASTAR}{p}'].sum(axis=0)
+            error[p] = 0.5 * sqrt((row[f'{ERR}{BETASTAR}{p}'] ** 2).sum(axis=0))
+
+        # Compute the relative error for this IP
+        rel_error[ip] = 0.5 * \
+            sqrt(sum([(error[p] ** 2) / (average_beta[p] ** 2) for p in PLANES]))
+
+        # Get the effective beta
+        eff_beta[ip] = sqrt(reduce(mul, average_beta.values()))
+
+    # Compute the whole relative error
+    relative_error = sum(rel_error.values())
+
+    # And get the imbalance
+    imbalance = eff_beta[IPS[0]] / eff_beta[IPS[1]]
+
+    return {'imbalance': imbalance,
+            'relative_error': relative_error,
+            'eff_beta_ip1': eff_beta[IPS[0]],
+            'rel_error_ip1': rel_error[IPS[0]],
+            'eff_beta_ip5': eff_beta[IPS[1]],
+            'rel_error_ip5': rel_error[IPS[1]]
+            }
+
+
+def print_luminosity(res):
+    LOG.info(f'{"Luminosity Imbalance":22s}: {res["imbalance"]}')
+    LOG.info(f'{"Relative error":22s}: {res["relative_error"]}')
+    LOG.info(f'{"Effective beta IP1":22s}: {res["eff_beta_ip1"]}')
+    LOG.info(f'{"Rel error IP1":22s}: {res["rel_error_ip1"]}')
+    LOG.info(f'{"Effective beta IP5":22s}: {res["eff_beta_ip5"]}')
+    LOG.info(f'{"Rel error IP5":22s}: {res["rel_error_ip5"]}')
+
+
+def get_significant_digits(res):
+    res['imbalance'], res['relative_error'] = significant_digits(
+            res['imbalance'], res['relative_error'])
+    res['eff_beta_ip1'], res['rel_error_ip1'] = significant_digits(
+            res['eff_beta_ip1'], res['rel_error_ip1'])
+    res['eff_beta_ip5'], res['rel_error_ip5'] = significant_digits(
+            res['eff_beta_ip5'], res['rel_error_ip5'])
+    return res
+
+
+def merge_tfs(directories, filename):
```

Comment:
You will be losing the headers if you `.append()` TfsDataFrames. Are we sure this is not a problem here?